src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexBoyerMoore.cs

```diff
@@ -43,15 +43,7 @@ public RegexBoyerMoore(string pattern, bool caseInsensitive, bool rightToLeft, C
             if (caseInsensitive)
             {
                 pattern = string.Create(pattern.Length, (pattern, culture), (span, state) =>
```

Comment:
You do not need to call `string.Create` at all here now. Just call `pattern.ToLower(culture)`

Reply:
right, missed that. thanks

Reply:
I was able to completely get rid of the call as it is already lowered by the RegexParser. I changed a few test cases to have more hits of the combination of a prefix and the IgnoreCase option and added an assert.